datalad_neuroimaging/extractors/fslfeat.py

```diff
@@ -0,0 +1,276 @@
+                rec = evs.get(con, {})
+                vec = rec.get(name, [0] * valid_props['evs_orig'])
+            else:
+                rec = contrasts.get(con, {})
+                vec = rec.get(
+                    name,
+                    [0] * valid_props['evs_real' if '_real' in name else 'evs_orig'])
+            vec[pos - 1] = val
+            rec[name] = vec
+            if name == 'ortho':
+                evs[con] = rec
+            else:
+                contrasts[con] = rec
+            continue
+        if var.startswith('conname_'):
+            # contrast names
+            con = int(var.split('.')[-1])
+            rec = contrasts.get(con, {})
+            rec['name'] = val
+            contrasts[con] = rec
+            continue
+
+        store_in[var] = val
+    # clean-up
+    for ev in evs.values():
+        # shed orthogonalization cfg, if noe was performed
+        if 'ortho' in ev and all(i == 0 for i in ev['ortho']):
+            ev.pop('ortho')
+    props['ev'] = [evs[i] for i in sorted(evs, key=lambda x: int(x))]
+    props['contrasts'] = [contrasts[i]
+                          for i in sorted(contrasts, key=lambda x: int(x))]
+    # TODO filter out all variables related to steps that were not
+    # performed
+    props['steps_disabled'] = [s if not s.endswith('_yn') else s[:-3]
+                               for s in sorted(disabled)]
+
+    # TODO make absolute paths relative to the dataset root
+    return props
+
+
+class MetadataExtractor(BaseMetadataExtractor):
+
+    _unique_exclude = {
+    }
+
+    def get_metadata(self, dataset, content):
+        analyses = []
+        log_progress(
+            lgr.info,
+            'extractorfslfeat',
+            'Start FSL Feat metadata extraction from %s', self.ds,
+            total=len(self.paths),
+            label='FSL Feat metadata extraction',
+            unit=' Files',
+        )
+        for f in self.paths:
+            absfp = opj(self.ds.path, f)
+            log_progress(
+                lgr.info,
+                'extractorfslfeat',
+                'Extract FSL Feat metadata from %s', absfp,
+                update=1,
+                increment=True)
+
+            if not basename(f) == 'design.fsf':
+                # ignore everything else. any analysis must have this, even if
+                # it never ran and has no results
+                lgr.debug("Ignoring non-analysis configuration %s", f)
+                continue
+
+            analysis_path = dirname(f)
+            analysis = {
+                'path': analysis_path,
+            }
+            analysis.update(_read_fsf(absfp, self.ds.path))
+
+            # look for per-contrast result tables
+            for i, rec in enumerate(analysis.get('contrasts', [])):
+                clustertable_fname = opj(
+                    self.ds.path, analysis_path, 'cluster_zstat{}.txt'.format(i + 1))
+                if exists(clustertable_fname):
+                    import pandas as pd
```

Comment:
IMHO quite heavy of a dependency, but oh well I guess.

Reply:
FTR: pandas is a core dependency of pybids. Heavy, but we already carry it, hence there is no penalty.